primitives/arithmetic/src/biguint.rs

```diff
@@ -585,9 +585,14 @@ pub mod tests {
 		let a = SHIFT / 2;
 		let b = SHIFT * 3 / 2;
 		let num: Double = 1 << a | 1 << b;
-		// example when `Single = u8`
-		// assert_eq!(num, 0b_0001_0000_0001_0000)
+		assert_eq!(num, 0x_0001_0000_0001_0000);
 		assert_eq!(split(num), (1 << a, 1 << a));
+
+		let a = SHIFT / 2 + 4;
```

Comment:
Can you adda a const assertion to make sure that this is correct? at the top of the code, SHIFT is defined as the difference between the bits of `Single` and `Double`. We can check this with const_assert now and `std::mem::size_of`

Reply:
L587 test not understand
and new test

Reply:
add shift_check